jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/OdeTimJsonTopology.java

```diff
@@ -66,48 +79,73 @@ public boolean isRunning() {
   /**
    * Builds a Kafka Streams topology for processing TIM JSON data.
    *
-   * @param topic the Kafka topic from which TIM JSON data is consumed and used to build the
-   *              topology.
    * @return the constructed Kafka Streams topology.
    */
-  public Topology buildTopology(String topic) {
+  public Topology buildTopology() {
     StreamsBuilder builder = new StreamsBuilder();
+    ObjectMapper objectMapper = new ObjectMapper();
 
-    // Create a windowed store with a retention period of 1 hour
-    KStream<String, String> timStream = builder.stream(topic);
-
-    timStream.groupByKey()
-        .windowedBy(TimeWindows.ofSizeWithNoGrace(Duration.ofHours(1))) // 1 hour window
-        .reduce(
-            (aggValue, newValue) -> newValue, // only keep latest value
-            Materialized.<String, String, WindowStore<Bytes, byte[]>>as("timjson-windowed-store")
-                .withRetention(Duration.ofHours(1))
-                .withKeySerde(Serdes.String())
+    // Read as KStream, filter to only include TMC generated messages, then convert
+    // to KTable
+    builder.stream(this.topic, Consumed.with(Serdes.String(), Serdes.String()))
+        .filter((key, value) -> {
+          try {
+            // Null values are tombstones, which are needed to remove entries from the
+            // KTable
+            if (value == null) {
+              return true;
+            }
+            // Parse the JSON and check if the message was generated by the TMC
+            JsonNode nodeMetadata = objectMapper.readTree(value).get("metadata");
+            boolean isTMCGenerated = "TMC".equals(nodeMetadata.get("recordGeneratedBy").asText());
+            log.debug("TMC Generated: {}, Key: {}", isTMCGenerated, key);
+            return isTMCGenerated;
+          } catch (Exception e) {
+            // Don't include unexpected JSON formats in the KTable
+            // This is only intended to be a graceful handling of unexpected JSON so logging
+            // is set to debug
+            log.debug("TIM topology failed to parse JSON for key {}: {}", key, e.getMessage());
+            return false;
+          }
+        }).toTable(
```

Comment:
suggestion: Why not use a GlobalKTable instead of having a unique app id to trick it into treating this as a global Ktable?  By replacing the `toTable` with something like the following:

```java
        }).to("filtered-topic-name");
    
    builder.globalTable("filtered-topic-name",
        Consumed.with(Serdes.String(), Serdes.String()),
        Materialized.<String, String, KeyValueStore<Bytes, byte[]>>as("timjson-ktable-store")
          .withKeySerde(Serdes.String()).withValueSerde(Serdes.String()));
```